### PR TITLE
Add & use useMemoValue

### PR DESCRIPTION
**Pull request by jacogr**
Inspiration from https://github.com/polkadot-js/apps/pull/8490 (which is now defunct)

We will probably not merge this, just pull pieces from it like we have done in #8750 (since each and every one needs some extensive testing to ensure it is still on-spec and doesn't create negative effects with any areas)

### CODE REVIEW

File: packages/react-hooks/src/ctx/ApiStats.tsx [hunk 1]
```diff
@@ -67,6 +64,16 @@ function getStats (...apis: ApiPromise[]): { stats: ProviderStats, when: number
...
 export const ApiStatsCtx = React.createContext<ApiStats[]>(EMPTY_STATE);
 
+function addStats (prev: ApiStats[], curr: ApiStats): ApiStats[] {
+  if (prev.length === 0) {
+    return [curr];
+  }
+
+  return prev.length === MAX_NUM
+    ? prev.concat(curr).slice(-MAX_NUM)
+    : prev.concat(curr);
```
Comment (jacogr) [inline packages/react-hooks/src/ctx/ApiStats.tsx:74]:
```suggestion
  return prev.length === 0
    ? [curr]
    : prev.length === MAX_NUM
      ? prev.concat(curr).slice(-MAX_NUM)
      : prev.concat(curr);
```

File: packages/react-hooks/src/useAccountInfo.ts [hunk 2]
```diff
@@ -247,7 +247,7 @@ function useAccountInfoImpl (value: string | null, isContract = false): UseAccou
 
   const isEditing = useCallback(() => isEditingName || isEditingTags, [isEditingName, isEditingTags]);
 
-  return useMemo(() => ({
+  return {
```
Comment (jacogr) [inline packages/react-hooks/src/useAccountInfo.ts:250]:
```suggestion
  return useMemo(() => ({
```

File: packages/react-hooks/src/useEventTrigger.ts [hunk 3]
```diff
@@ -33,6 +33,7 @@ function useEventTriggerImpl (checks: EventCheck[], filter: (record: EventRecord
   const memoChecks = useMemoValue(checks);
   const mountedRef = useIsMountedRef();
   const eventRecords = useCall<Vec<EventRecord>>(api.query.system.events);
+  const memoChecks = useMemoValue(checks);
```
Comment (jacogr) [inline packages/react-hooks/src/useEventTrigger.ts:36]:
```suggestion
  const memoChecks = useMemoValue(checks);
```

Added above.

File: packages/react-hooks/src/useExtrinsicTrigger.ts [hunk 4]
```diff
@@ -20,6 +20,7 @@ function useExtrinsicTriggerImpl (checks: ExtrinsicCheck[]): string {
   const mountedRef = useIsMountedRef();
   const memoChecks = useMemoValue(checks);
   const block = useCall<SignedBlockExtended>(api.derive.chain.subscribeNewBlocks);
+  const memoChecks = useMemoValue(checks);
```
Comment (jacogr) [inline packages/react-hooks/src/useExtrinsicTrigger.ts:23]:
```suggestion
```

Added above.
